files/install_tool_shed_tools.py

```diff
@@ -547,40 +612,9 @@ def install_tools(options):
                        tool['tool_panel_section_id'] or tool['tool_panel_section_label'],
                        tool['revision'], dt.datetime.now() - istart))
             try:
-                response = tsc.install_repository_revision(
-                    tool['tool_shed_url'], tool['name'], tool['owner'],
-                    tool['revision'], tool['install_tool_dependencies'],
-                    tool['install_repository_dependencies'],
-                    tool['tool_panel_section_id'],
-                    tool['tool_panel_section_label'])
-                tool_id = None
-                tool_status = None
-                if len(response) > 0 and isinstance(response, list):
-                    tool_status = response[-1].get('status', None)
-                    tool_id = response[-1].get('id', None)
-                elif isinstance(response, dict) and response.get('status', None) == 'ok':
-                    # This rare case happens if a tool is already installed but
-                    # was not recognised as such in the above check. In such a
-                    # case the return value looks like this:
-                    # {u'status': u'ok', u'message': u'No repositories were
-                    #  installed, possibly because the selected repository has
-                    #  already been installed.'}
-                    log.debug("\tTool {0} is already installed.".format(tool['name']))
-                if tool_id and tool_status:
-                    # Possibly an infinite loop here. Introduce a kick-out counter?
-                    log.debug("\tTool installing", extra={'same_line': True})
-                    while tool_status not in ['Installed', 'Error']:
-                        log.debug("", extra={'same_line': True})
-                        time.sleep(10)
-                        tool_status = update_tool_status(tsc, tool_id)
-                    end = dt.datetime.now()
-                    installed_tools.append({'name': tool['name'], 'owner': tool['owner'],
-                                            'revision': tool['revision']})
-                    log.debug("\tTool %s installed successfully (in %s) at revision %s"
-                              % (tool['name'], str(end - start), tool['revision']))
-                else:
-                    end = dt.datetime.now()
-                    log.error("\tCould not retrieve tool status for {0}".format(tool['name']))
+                response = install_repository_revision(tool, tsc)
```

Comment:
It seems that bioblend's install_repository_revision() only returns when the installation process is finished, so the update process would never get called. Effectively the progress status was not functional before.